pkgs/development/libraries/mpich/default.nix

```diff
@@ -29,21 +42,38 @@ stdenv.mkDerivation  rec {
 
   outputs = [ "out" "doc" "man" ];
 
-  configureFlags = [
-    "--enable-shared"
-    "--with-pm=${withPmStr}"
-  ] ++ lib.optionals (lib.versionAtLeast gfortran.version "10") [
-    "FFLAGS=-fallow-argument-mismatch" # https://github.com/pmodels/mpich/issues/4300
-    "FCFLAGS=-fallow-argument-mismatch"
-  ] ++ lib.optionals pmixSupport [
-    "--with-pmix=${lib.getDev pmix}"
-  ];
+  configureFlags =
+    [
+      "--enable-shared"
+      "--with-pm=${withPmStr}"
+    ]
+    ++ lib.optionals (lib.versionAtLeast gfortran.version "10") [
+      "FFLAGS=-fallow-argument-mismatch" # https://github.com/pmodels/mpich/issues/4300
+      "FCFLAGS=-fallow-argument-mismatch"
+    ]
+    ++ lib.optionals withSlurm [
+      "--with-pmi=pmi2"
+      "--with-pmilib=slurm"
+    ]
+    ++ lib.optionals pmixSupport [ "--with-pmix=${lib.getDev pmix}" ];
 
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ gfortran python3 ];
-  buildInputs = [ perl openssh hwloc ]
-    ++ lib.optional (!stdenv.isDarwin) ch4backend;
+  nativeBuildInputs = [
+    gfortran
+    python3
+  ];
+  buildInputs =
+    [
+      perl
+      openssh
+      hwloc
+    ]
+    ++ lib.optionals withSlurm [
+      (lib.getLib slurm)
+      (lib.getDev slurm)
```

Comment:
Do you really need both? `slurm` is not enough?

Reply:
Alack, it's not. Autoconf fails to infer the lib's location from `.dev`